spec/App/MarkdownSpec.php

```diff
@@ -21,11 +21,32 @@
 
 use PhpSpec\ObjectBehavior;
 use Prophecy\Argument;
+use App\Writer;
+use App\Reader;
 
 class MarkdownSpec extends ObjectBehavior
 {
-    function it_is_initializable()
+    function it_outputs_converted_text(Writer $writer)
     {
-        $this->shouldHaveType('App\Markdown');
+        // Pass object into constructor
+        $this->beConstructedWith($writer);
+
+        // Describe method App\Writer::writeText() should called before example
+    	$writer->writeText('Hi, there')->shouldBeCalled();
+
+    	// when
+        $this->outputHtml('Hi, there', $writer);
+    }
+
+    function it_converts_text_from_an_external_source(Reader $reader)
```

Comment:
need to pass `Writer` too as it needed in `createForWriting` as I commented in https://github.com/sitrunlab/learn-phpspec/pull/2/files#r41588508